src/pages/payments/edit/hooks/useTabs.ts

```diff
@@ -20,31 +25,59 @@ interface Params {
 export function useTabs(params: Params) {
   const [t] = useTranslation();
 
+  const { isAdmin, isOwner } = useAdmin();
+
+  const hasPermission = useHasPermission();
+  const entityAssigned = useEntityAssigned();
+
   const { id } = useParams();
 
   const { payment } = params;
 
+  const canOnlyEdit = hasPermission('edit_payment') || entityAssigned(payment);
+
+  const canEditAndView =
+    hasPermission('view_payment') ||
+    hasPermission('edit_payment') ||
+    entityAssigned(payment);
+
   let tabs: Tab[] = [
     {
       name: t('edit'),
       href: route('/payments/:id/edit', { id }),
     },
-    {
-      name: t('apply'),
-      href: route('/payments/:id/apply', { id }),
-    },
-    {
-      name: t('refund'),
-      href: route('/payments/:id/refund', { id }),
-    },
-    {
-      name: t('documents'),
-      href: route('/payments/:id/documents', { id }),
-    },
-    {
-      name: t('custom_fields'),
-      href: route('/payments/:id/payment_fields', { id }),
-    },
+    ...(canOnlyEdit
+      ? [
+          {
+            name: t('apply'),
+            href: route('/payments/:id/apply', { id }),
+          },
+        ]
+      : []),
```

Comment:
Honestly, this looks super messy. Is it possible to abstract this or integrate logic directly in the tabs component?

Reply:
@beganovich Yeah, you are right. I just formatted it to make the code look better. Let me know your thoughts.

Reply:
Honestly, I thought you were going to implement `enabled` property on tab, seems like reasonable thing to do?